### PR TITLE
Add getSchoolByCountry

### PR DESCRIPTION
**Pull request by brian386**
## Notion ticket link

[Get schools by country](https://www.notion.so/uwblueprintexecs/47bf4727a2444bccad4a91952898b8f3?v=9658436bf9be409e90581631a7155f16&p=0a33103357834cc890e6d403fb7af18e)

## Implementation description
* Added a getSchoolByCountry function
* added 2 test cases: 1 for valid country and 1 for invalid country

## Steps to test
1.

## What should reviewers focus on?
* 

## Checklist
- [ ] My PR name is descriptive and in imperative tense
- [ ] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [ ] I have run the appropriate linter(s)
- [ ] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: backend/testUtils/school.ts [hunk 1]
```diff
@@ -41,6 +41,42 @@ export const testSchools = [
   },
 ];
 
+// set up test schools
+export const testSchools2 = [
```
Comment (mahadzaryab1) [inline backend/testUtils/school.ts:45]: why did we make a new test schools variable?

File: backend/services/implementations/schoolService.ts [hunk 2]
```diff
@@ -52,6 +52,26 @@ class SchoolService implements ISchoolService {
...
+   */
+  async getSchoolByCountry(country: string): Promise<SchoolResponseDTO[]> {
+    let schools: Array<School> | null;
+
+    try {
+      schools = await MgSchool.find({ country });
+
+      if (!schools.length) {
+        throw new Error(`Country ${country} not found`);
+      }
```
Comment (mahadzaryab1) [inline backend/services/implementations/schoolService.ts]: @joyce-shi - should we be throwing errors here (and in the other services), if we don't find anything? I feel like we could just return an empty array here and the upstream caller can do whatever it wants with it.

**Review by mahadzaryab1 — COMMENTED** [review]
added a couple of comments

File: backend/testUtils/school.ts (hunk 1 above)
Comment (brian386) [reply backend/testUtils/school.ts:45]: I wanted to have a new set of test schools with different countries to make sure the filtering works. Should I combine them into the existing test school array? I'm scared doing so might break the older test cases

Comment (carissa-tang) [reply backend/testUtils/school.ts:45]: you could update the original `testSchools` to have `some-country1` and `some-country2` and that shouldn't break anything. you could also add a third entry with `some-country1` again. just update the existing tests if something fails

File: backend/services/implementations/schoolService.ts (hunk 2 above)
Comment (joyce-shi) [reply backend/services/implementations/schoolService.ts]: +1 to that. I think the most likely use case for this is that admin staff would be selecting from a pre-populated dropdown to query by country. Given this, no error throwing necessary.

File: backend/testUtils/school.ts (hunk 1 above)
Comment (carissa-tang) [reply backend/testUtils/school.ts:45]: nvm we can keep separate variables for more clarity? i think that's what we're going for based on other PRs

Comment (brian386) [reply backend/testUtils/school.ts:45]: okay, sounds good!

File: backend/services/interfaces/schoolService.ts [hunk 3]
```diff
@@ -61,6 +61,14 @@ export interface ISchoolService {
    */
   getSchoolsBySubregion(subRegion: string): Promise<SchoolResponseDTO[]>;
 
+  /**
+   * This method retrieve all Schools with the given country
+   * @param country the country the school is located in
+   * @returns returns array of requested SchoolResponseDTO
+   * @throws Error if retrieval fails
+   */
+  getSchoolsBySubregion(country: string): Promise<SchoolResponseDTO[]>;
```
Comment (carissa-tang) [inline backend/services/interfaces/schoolService.ts]:
```suggestion
  getSchoolsByCountry(country: string): Promise<SchoolResponseDTO[]>;
```

File: backend/services/implementations/schoolService.ts [hunk 4]
```diff
@@ -52,6 +52,21 @@ class SchoolService implements ISchoolService {
     }
   }
 
+  /**
+   * This method gets all schools with the given country from the database.
+   */
+  async getSchoolByCountry(country: string): Promise<SchoolResponseDTO[]> {
```
Comment (carissa-tang) [inline backend/services/implementations/schoolService.ts]:
```suggestion
  async getSchoolsByCountry(country: string): Promise<SchoolResponseDTO[]> {
```
nit: can you rename the function to school**s**

File: backend/services/implementations/schoolService.ts [hunk 5]
```diff
@@ -52,6 +52,21 @@ class SchoolService implements ISchoolService {
...
+    let schools: Array<School> | null;
+
+    try {
+      schools = await MgSchool.find({ country });
+      return await this.mapSchoolsToSchoolResponseDTOs(schools);
+    } catch (error: unknown) {
+      Logger.error(`Failed to get schools. Reason = ${getErrorMessage(error)}`);
```
Comment (carissa-tang) [inline backend/services/implementations/schoolService.ts]:
```suggestion
      Logger.error(`Failed to get schools for country ${country}. Reason = ${getErrorMessage(error)}`);
```

**Review by carissa-tang — CHANGES_REQUESTED** [review]

**Review by carissa-tang — APPROVED** [review]
lgtm!

**Pull request merged**